engine/classes/Elgg/Database/MetastringsTable.php

```diff
@@ -112,8 +112,11 @@ private function getIdCaseSensitive($string) {
 		$string = (string)$string;
 		return $this->cache->get($string, function() use ($string) {
 			$escaped_string = $this->db->sanitizeString($string);
-			$query = "SELECT id FROM {$this->getTableName()} WHERE string = BINARY '$escaped_string' LIMIT 1";
-			$results = $this->db->getData($query);
+			$query = "SELECT id FROM {$this->getTableName()} WHERE string = BINARY :string LIMIT 1";
+			$params = [
```

Comment:
When converting to params, eliminate the sanitize call. FWIW I prefer the other syntax for up to a couple params:
```php
$query = "SELECT id FROM {$this->getTableName()} WHERE string = BINARY ? LIMIT 1";
$results = $this->db->getData($query, null, [$string]);
```